spec/controllers/api/v1/changes_controller_spec.rb

```diff
@@ -0,0 +1,118 @@
+
+require 'rails_helper'
+# Create
+RSpec.describe "Api::V1::Changes", type: :request do
+  describe "POST /api/v1/changes" do
+    let(:valid_params) do
+      {
+        user_id: 1,
+        old: { name: "Bruce Norries" },
+        new: { name: "Bruce Willis" }
+      }
+    end
+
+    let(:invalid_user_id_params) do
+      {
+        user_id: -1,
+        old: { name: "Bruce Norries" },
+        new: { name: "Bruce Willis" }
+      }
+    end
+
+    let(:empty_data_params) do
+      {
+        user_id: 1,
+        old: {},
+        new: {}
+      }
+    end
+
+    it "tracks changes successfully with valid data" do
+      post "/api/v1/changes", params: valid_params
+      expect(response).to have_http_status(:created)
+      expect(JSON.parse(response.body)['message']).to eq('Changes tracked successfully')
+    end
+
+    it "returns error for invalid user_id" do
+      post "/api/v1/changes", params: invalid_user_id_params
+      expect(response).to have_http_status(:unprocessable_entity)
+      expect(JSON.parse(response.body)['error']).to eq('user_id must be a positive integer')
+    end
+
+    it "returns error when both old and new data are empty" do
+      post "/api/v1/changes", params: empty_data_params
+      expect(response).to have_http_status(:bad_request)
+      expect(JSON.parse(response.body)['error']).to eq('Both old and new data cannot be empty')
+    end
+
+    it "returns error for unexpected issues" do
+      allow(Changes::Tracker).to receive(:call).and_raise(StandardError)
+      post "/api/v1/changes", params: valid_params
+      expect(response).to have_http_status(:internal_server_error)
+      expect(JSON.parse(response.body)['error']).to eq('An unexpected error occurred')
+    end
+  end
+end
+
+# Index
+RSpec.describe "Api::V1::Changes", type: :request do
+  describe "GET /api/v1/changes" do
+    before do
+      # Create test changes
+      Change.create!(user_id: 1, field: "name", old_value: "Bruce Norries", new_value: "Bruce Willis", changed_at: "2024-08-10")
```

Comment:
does this data remain in the database or is it always cleaned before starting tests?

Reply:
Forgot to answer this one. No it does not remain in the database i think by default the transaction starts when the test begins and is rolled back after the test completes